src/adagrams.js

```diff
@@ -1,15 +1,229 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterPool = [
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "B",
+    "B",
+    "C",
+    "C",
+    "D",
+    "D",
+    "D",
+    "D",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "F",
+    "F",
+    "G",
+    "G",
+    "G",
+    "H",
+    "H",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "J",
+    "K",
+    "L",
+    "L",
+    "L",
+    "L",
+    "M",
+    "M",
+    "N",
+    "N",
+    "N",
+    "N",
+    "N",
+    "N",
+    "O",
+    "O",
+    "O",
+    "O",
+    "O",
+    "O",
+    "O",
+    "O",
+    "P",
+    "P",
+    "Q",
+    "R",
+    "R",
+    "R",
+    "R",
+    "R",
+    "R",
+    "S",
+    "S",
+    "S",
+    "S",
+    "T",
+    "T",
+    "T",
+    "T",
+    "T",
+    "T",
+    "U",
+    "U",
+    "U",
+    "U",
+    "V",
+    "V",
+    "W",
+    "W",
+    "X",
+    "Y",
+    "Y",
+    "Z",
+  ];
+
+  const shuffledLetterPool = letterPool.sort(() => Math.random() - 0.5);
+
+  let hand = shuffledLetterPool.slice(0, 10);
+
+  return hand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  let loop_count = 0;
+  for (let letter of input) {
+    for (let handLetter of lettersInHand) {
+      if (letter.toUpperCase() == handLetter.toUpperCase()) {
+        loop_count++;
+        const index = lettersInHand.indexOf(handLetter);
+        if (index > -1) {
+          lettersInHand.splice(index, 1);
+        }
+      }
+    }
+  }
+  if (loop_count === input.length) {
+    return true;
+  } else {
+    return false;
+  }
```

Comment:
The approach taken here definitely works! When writing code, it's always important to consider its complexity. It's a good idea to ask oneself, is there a simpler approach? For example, if you make a shallow copy of the `lettersInHand` argument, and perform the check to see whether a letter from the `input` argument is in the `lettersInHand` and remove it if so, otherwise if you find a letter that's not in the hand, immediately return false, you could simplify the code to something like: 
```javascript
const lettersInHandCopy = [...lettersInHand];
for (const letter in input.toUpperCase()) {
  if (lettersInHandCopy.includes(letter)) {
      const index = lettersInHandCopy.indexOf(letter);
      lettersInHandCopy.splice(index, 1);
  } else {
    // we've found a letter not in the hand, so we don't need to check anything else!
    return false;
  }
  // if we haven't returned yet, then all the letters are in the hand, so we can return true!
  return true;
}
```